lib/customer.rb

```diff
@@ -0,0 +1,48 @@
+require 'csv'
+require_relative '../lib/arg_error'
+
+class Customer
+  include ArgError
+
+  attr_reader :id
+  attr_accessor :email, :address
+
+  def initialize(id, email, address)
+    arg_class_check(id, "id", Integer)
+    arg_class_check(email, "email", String)
+    arg_class_check(address, "address", Hash)
```

Comment:
This input validation will prove especially handy when we get into web development shortly!